std/jwt/mod.ts

```diff
@@ -0,0 +1,194 @@
+import type { Algorithm } from "./algorithm.ts";
+import * as base64url from "../encoding/base64url.ts";
+import { encodeToString as convertUint8ArrayToHex } from "../encoding/hex.ts";
+import {
+  create as createSignature,
+  verify as verifySignature,
+} from "./signature.ts";
+import { verify as verifyAlgorithm } from "./algorithm.ts";
+/*
+ * JWT §4.1: The following Claim Names are registered in the IANA
+ * "JSON Web Token Claims" registry established by Section 10.1. None of the
+ * claims defined below are intended to be mandatory to use or implement in all
+ * cases, but rather they provide a starting point for a set of useful,
+ * interoperable claims.
+ * Applications using JWTs should define which specific claims they use and when
+ * they are required or optional.
+ */
+export interface PayloadObject {
+  iss?: string;
+  sub?: string;
+  aud?: string[] | string;
+  exp?: number;
+  nbf?: number;
+  iat?: number;
+  jti?: string;
+  [key: string]: unknown;
+}
+
+export type Payload = PayloadObject | string;
+
+/*
+ * JWS §4.1.1: The "alg" value is a case-sensitive ASCII string containing a 
+ * StringOrURI value. This Header Parameter MUST be present and MUST be
+ * understood and processed by implementations.
+ */
+export interface Header {
+  alg: Algorithm;
+  [key: string]: unknown;
+}
+
+const encoder = new TextEncoder();
+const decoder = new TextDecoder();
+
+function isExpired(exp: number, leeway = 0): boolean {
+  return exp + leeway < Date.now() / 1000;
+}
+
+/*
+ * Helper function: setExpiration()
+ * returns the number of seconds since January 1, 1970, 00:00:00 UTC
+ */
+export function setExpiration(exp: number | Date): number {
+  return Math.round(
+    (exp instanceof Date ? exp.getTime() : Date.now() + exp * 1000) / 1000,
+  );
+}
+
+function tryToParsePayload(input: string) {
+  try {
+    return JSON.parse(input);
+  } catch {
+    return input;
+  }
+}
+
+export function decode(
```

Comment:
For a public API, we should have some JSDoc (with examples) and a "concrete" return value...  I am also surprised we can't have a narrow type than `unknown`.

Reply:
Ok, I see you have `isTokenObject()` but the examples in the README don't use this.  Why would we want to potentially not validate the token object as part of the decode... no one wants an unvalidated object do they?

Reply:
@timonson and I had a discussion about that. @timonson can you expand on that?

Reply:
We didn't have `isTokenObject` from the beginning and we wanted to separate `verify` from `decode` so that people could use `decode` in cases where the 'no `crit` header' error is thrown; in the sense of a *helper function*. I can see that the separation now is not really necessary anymore but at the same time both functions fulfill one clearly defined task respectively. So it doesn't hurt to keep them, I guess, but @kitsonk is right, it is not necessary to separate them anymore.

Reply:
By the way, I just made changes to `isTokenObject`. I would love to get feedback on that.

Reply:
@kitsonk we moved `isTokenObject` to `decode` now. What do you think?